vscode/src/commands/context/git-api.ts

```diff
@@ -0,0 +1,152 @@
+import * as vscode from 'vscode'
+
+import { type ContextItem, ContextItemSource, TokenCounter, displayPath } from '@sourcegraph/cody-shared'
+import { logError } from '../../log'
+import type { Repository } from '../../repository/builtinGitExtension'
+
+/**
+ * Generates context files from the git diff and git log of a given repository.
+ *
+ * @param gitRepo - The git repository object.
+ * @param template - An optional template for a git commit context file.
+ * @returns A promise that resolves to an array of ContextItem objects representing the git diff and git log.
+ * @throws If there is an error retrieving the git diff or git log.
+ */
+export async function getContextFilesFromGitApi(
+    gitRepo: Repository,
+    template?: string
+): Promise<ContextItem[]> {
+    const [diffContext, logContext] = await Promise.all([
+        getContextFilesFromGitDiff(gitRepo),
+        getContextFilesFromGitLog(gitRepo),
+    ])
+
+    const contextItems = [...diffContext, ...logContext]
+    if (template) {
+        contextItems.push(getGitCommitTemplateContextFile(template))
+    }
+    return contextItems
+}
+
+/**
+ * Generate context files from the git diff of a given repository.
+ *
+ * @param gitRepo - The git repository object.
+ * @returns A promise that resolves to an array of ContextItem objects representing the git diff.
+ * @throws If the git diff output is empty or if there is an error retrieving the git diff.
+ */
+async function getContextFilesFromGitDiff(gitRepo: Repository): Promise<ContextItem[]> {
+    try {
+        const diffOutput = await gitRepo?.diff(true)
```

Comment:
```suggestion
        const hasStagedChanges = gitRepo?.state?.indexChanges?.length
        const diffOutput = await gitRepo?.diff(hasStagedChanges)
```

@RXminuS @umpox this is what we need to change if we've decided to generate commit for unstaged files if there is no staged files